### PR TITLE
Add vendoring with govendor

### PR DESCRIPTION
**Pull request by moorereason**
Doesn't check in vendored packages.  Rewrites Makefile to use govendor helpers.

### CODE REVIEW

**Comment by bep** [comment]
The vendor.json file; is that a standard format?

**Comment by moorereason** [comment]
There is no standard format.  That's govendor's format.  The only standard is that the Go tools look for vendored packages in the `vendor/` directory.

**Comment by moorereason** [comment]
Tagged for 0.17.  FIxes #2119.

**Comment by bep** [comment]
I merged this as it does solve the problem ... but we should do an evaluation of the vendoring options for 0.18.

**Comment by nathany** [comment]
When I read the [release notes](https://github.com/spf13/hugo/releases/tag/v0.17) for 0.17, I just expected the code under vendor/ to be checked in.

Should `make govendor` be mentioned in the README?

File: Makefile [hunk 1]
```diff
@@ -34,39 +32,45 @@ docker:
 	docker cp hugo-build:/go/bin/hugo .
 	docker rm hugo-build
 
+govendor:
+	go get -u github.com/kardianos/govendor
+	go install github.com/kardianos/govendor
```
Comment (nathany) [inline Makefile]: FYI

> "Get downloads the packages named by the import paths, along with their
dependencies. It then installs the named packages, like 'go install'." - `go get -h`

**Comment by moorereason** [comment]
The Makefile has already been fixed in master.

And yes, we may need to update the README.

File: Makefile (hunk 1 above)
Comment (nathany) [inline Makefile]: Which is to say the `go install` line is superfluous. 

It still appears here atm: https://github.com/spf13/hugo/blob/master/Makefile#L37

Comment (moorereason) [inline Makefile]: You're right.  I was looking at this on my phone and misunderstood where your comment was in the Makefile.  The `go install` does indeed need to be removed.

**Pull request merged**